### ISSUE TITLE
Workloads are not removed on app deletion in k8s v1.16

### ISSUE THREAD

**Issue by gitlawr** [rancher/rancher#22684]
**Rancher Version**
v2.3.0-rc1

**Steps to reproduce (least amount of steps as possible):**
1. Go to `Global - Catalogs`, add a catalog from `https://github.com/gitlawr/helm-charts.git, branch: k8s-1.16`(Current community stable charts are not compatible with k8s v1.16, so this is a stable mysql chart that tweaks the deployment resource for k8s v1.16)
2. Install an app with `persistence.enabled=false` answers to the Default project. Wait till the app is installed and active.
3. Delete the app.

**Result:**
The app is deleted from the `apps` page, But the mysql deployment remains in the workload page.

**Other details:**
1. It works in a v1.15 k8s cluster.
2. I can apply and delete these resources successfully from kubectl(v1.13.1):
```
$ helm template ./stable/mysql --name=mysql --namespace=mysql \
    --set persistence.enabled=false | kubectl apply -f -
$ helm template ./stable/mysql --name=mysql --namespace=mysql \
    --set persistence.enabled=false | kubectl delete -f -
```
3. The resources in the chart include secret, service, and deployment. The secret and the service are deleted as expected.
4. There are no errors but warnings from rancher server logs:
```
2019/09/06 07:53:37 [INFO] Creating roleBinding User u-wi7dorfzny Role edit
[main] 2019/09/06 07:53:39 Starting Tiller v2.10+unreleased (tls=false)
[main] 2019/09/06 07:53:39 GRPC listening on :41799
[main] 2019/09/06 07:53:39 Probes listening on :35662
[main] 2019/09/06 07:53:39 Storage driver is ConfigMap
[main] 2019/09/06 07:53:39 Max history per release is 1
[tiller] 2019/09/06 07:53:39 getting history for release mysql-from-chart
[storage] 2019/09/06 07:53:39 getting release history for "mysql-from-chart"
Release "mysql-from-chart" does not exist. Installing it now.
[tiller] 2019/09/06 07:53:39 preparing install for mysql-from-chart
[storage] 2019/09/06 07:53:39 getting release history for "mysql-from-chart"
[tiller] 2019/09/06 07:53:39 rendering mysql chart using values
2019/09/06 07:53:39 info: manifest "mysql/templates/configurationFiles-configmap.yaml" is empty. Skipping.
2019/09/06 07:53:39 info: manifest "mysql/templates/servicemonitor.yaml" is empty. Skipping.
2019/09/06 07:53:39 info: manifest "mysql/templates/pvc.yaml" is empty. Skipping.
2019/09/06 07:53:39 info: manifest "mysql/templates/initializationFiles-configmap.yaml" is empty. Skipping.
[tiller] 2019/09/06 07:53:39 performing install for mysql-from-chart
[tiller] 2019/09/06 07:53:39 executing 1 crd-install hooks for mysql-from-chart
[tiller] 2019/09/06 07:53:39 hooks complete for crd-install mysql-from-chart
[kube] 2019/09/06 07:53:39 warning: failed to load schema: SchemaError(io.k8s.api.core.v1.ComponentStatus): invalid object doesn't have additional properties
[tiller] 2019/09/06 07:53:39 executing 1 pre-install hooks for mysql-from-chart
[tiller] 2019/09/06 07:53:39 hooks complete for pre-install mysql-from-chart
[storage] 2019/09/06 07:53:39 getting release history for "mysql-from-chart"
[storage] 2019/09/06 07:53:39 creating release "mysql-from-chart.v1"
[storage] 2019/09/06 07:53:39 getting release history for "mysql-from-chart"
[kube] 2019/09/06 07:53:39 building resources from manifest
[kube] 2019/09/06 07:53:39 warning: failed to load schema: SchemaError(io.k8s.api.core.v1.ComponentStatus): invalid object doesn't have additional properties
[kube] 2019/09/06 07:53:39 creating 4 resource(s)
[tiller] 2019/09/06 07:53:39 executing 1 post-install hooks for mysql-from-chart
[tiller] 2019/09/06 07:53:39 hooks complete for post-install mysql-from-chart
[storage] 2019/09/06 07:53:39 updating release "mysql-from-chart.v1"
NAME:   mysql-from-chart
[storage] 2019/09/06 07:53:39 getting last revision of "mysql-from-chart"
[storage] 2019/09/06 07:53:39 getting release history for "mysql-from-chart"
[kube] 2019/09/06 07:53:39 warning: failed to load schema: SchemaError(io.k8s.api.core.v1.ComponentStatus): invalid object doesn't have additional properties
[kube] 2019/09/06 07:53:39 Doing get for Secret: "mysql-from-chart"
[kube] 2019/09/06 07:53:39 get relation pod of object: mysql-from-chart/Secret/mysql-from-chart
[kube] 2019/09/06 07:53:39 Doing get for ConfigMap: "mysql-from-chart-test"
[kube] 2019/09/06 07:53:40 get relation pod of object: mysql-from-chart/ConfigMap/mysql-from-chart-test
[kube] 2019/09/06 07:53:40 Doing get for Service: "mysql-from-chart"
[kube] 2019/09/06 07:53:40 get relation pod of object: mysql-from-chart/Service/mysql-from-chart
[kube] 2019/09/06 07:53:40 Doing get for Deployment: "mysql-from-chart"
[kube] 2019/09/06 07:53:40 get relation pod of object: mysql-from-chart/Deployment/mysql-from-chart
LAST DEPLOYED: Fri Sep  6 07:53:39 2019
NAMESPACE: mysql-from-chart
STATUS: DEPLOYED

RESOURCES:
==> v1/Secret
NAME              TYPE    DATA  AGE
mysql-from-chart  Opaque  2     1s

==> v1/ConfigMap
NAME                   DATA  AGE
mysql-from-chart-test  1     1s

==> v1/Service
NAME              TYPE       CLUSTER-IP   EXTERNAL-IP  PORT(S)   AGE
mysql-from-chart  ClusterIP  10.43.52.12  <none>       3306/TCP  1s

==> v1/Deployment
NAME              DESIRED  CURRENT  UP-TO-DATE  AVAILABLE  AGE
mysql-from-chart  1        1        1           0          1s

==> v1/Pod(related)
NAME                               READY  STATUS    RESTARTS  AGE
mysql-from-chart-5594bdb788-89p2x  0/1    Init:0/1  0         1s


NOTES:
MySQL can be accessed via port 3306 on the following DNS name from within your cluster:
mysql-from-chart.mysql-from-chart.svc.cluster.local

To get your root password run:

    MYSQL_ROOT_PASSWORD=$(kubectl get secret --namespace mysql-from-chart mysql-from-chart -o jsonpath="{.data.mysql-root-password}" | base64 --decode; echo)

To connect to your database:

1. Run an Ubuntu pod that you can use as a client:

    kubectl run -i --tty ubuntu --image=ubuntu:16.04 --restart=Never -- bash -il

2. Install the mysql client:

    $ apt-get update && apt-get install mysql-client -y

3. Connect using the mysql cli, then provide your password:
    $ mysql -h mysql-from-chart -p

To connect to your database directly from outside the K8s cluster:
    MYSQL_HOST=127.0.0.1
    MYSQL_PORT=3306

    # Execute the following command to route the connection:
    kubectl port-forward svc/mysql-from-chart 3306

    mysql -h ${MYSQL_HOST} -P${MYSQL_PORT} -u root -p${MYSQL_ROOT_PASSWORD}


[main] 2019/09/06 07:55:08 Starting Tiller v2.10+unreleased (tls=false)
[main] 2019/09/06 07:55:08 GRPC listening on :34998
[main] 2019/09/06 07:55:08 Probes listening on :46620
[main] 2019/09/06 07:55:08 Storage driver is ConfigMap
[main] 2019/09/06 07:55:08 Max history per release is 1
[storage] 2019/09/06 07:55:08 getting release history for "mysql-from-chart"
[tiller] 2019/09/06 07:55:08 uninstall: Deleting mysql-from-chart
[tiller] 2019/09/06 07:55:08 executing 1 pre-delete hooks for mysql-from-chart
[tiller] 2019/09/06 07:55:08 hooks complete for pre-delete mysql-from-chart
[storage] 2019/09/06 07:55:08 updating release "mysql-from-chart.v1"
[kube] 2019/09/06 07:55:08 warning: failed to load schema: SchemaError(io.k8s.api.core.v1.ComponentStatus): invalid object doesn't have additional properties
[kube] 2019/09/06 07:55:08 Starting delete for "mysql-from-chart" Service
[kube] 2019/09/06 07:55:08 warning: failed to load schema: SchemaError(io.k8s.api.core.v1.ComponentStatus): invalid object doesn't have additional properties
[kube] 2019/09/06 07:55:08 Starting delete for "mysql-from-chart" Deployment
[kube] 2019/09/06 07:55:08 Using reaper for deleting "mysql-from-chart"
[kube] 2019/09/06 07:55:08 the server could not find the requested resource
[kube] 2019/09/06 07:55:08 warning: failed to load schema: SchemaError(io.k8s.api.core.v1.ComponentStatus): invalid object doesn't have additional properties
[kube] 2019/09/06 07:55:08 Starting delete for "mysql-from-chart-test" ConfigMap
[kube] 2019/09/06 07:55:08 warning: failed to load schema: SchemaError(io.k8s.api.core.v1.ComponentStatus): invalid object doesn't have additional properties
[kube] 2019/09/06 07:55:08 Starting delete for "mysql-from-chart" Secret
[tiller] 2019/09/06 07:55:08 executing 1 post-delete hooks for mysql-from-chart
[tiller] 2019/09/06 07:55:08 hooks complete for post-delete mysql-from-chart
[tiller] 2019/09/06 07:55:08 purge requested for mysql-from-chart
[storage] 2019/09/06 07:55:08 deleting release "mysql-from-chart.v1"
2019/09/06 07:55:09 [WARNING] release "mysql-from-chart" deleted
```

**Comment by gitlawr** [comment]
Also, note that the warnings are from https://github.com/rancher/helm/blob/fba1e0f8cf6ef8cb814817ed8ba15ffd2c0d6526/pkg/kube/client.go#L188
```
[kube] 2019/09/06 07:55:08 warning: failed to load schema: SchemaError(io.k8s.api.core.v1.ComponentStatus): invalid object doesn't have additional properties

```

**Comment by gitlawr** [comment]
Similar logs are observed in https://github.com/rancher/rancher/issues/20959

**Comment by dramich** [comment]
This is caused by helm being on an old version of kubectl as seen [here](https://github.com/kubernetes/kubernetes/blob/release-1.10/pkg/kubectl/delete.go#L90). Deployment was just the first thing to break, any of the other APIs that were moved could have similar issues.

**Comment by dramich** [comment]
When testing this update it's important to validate a 1.15 cluster with the normal app testing scenarios (upgrade, rollback etc.) using our current catalog apps. 

When moving on to test a 1.16 cluster most of our apps will not work so a custom catalog will most likely need to be used and caution should be exercised when doing this as all of the failures I saw during this testing was due to the chart being broke in some way or another, not the actual helm uplift causing the problem.

**Comment by sangeethah** [comment]
1.16 compatible charts - https://github.com/rancher/charts/pull/272

**Comment by sowmyav27** [comment]
Verified using this PR: https://github.com/rancher/charts/pull/272 - branch.

- This version is NOT available - `etcd-operator v0.8.4` but `etcd-operator v0.9.0` worked.

- These apps fail to deploy:
```
prometheus chart v9.1.0
Magento v5.1.4
```

- Kafka fails to deploy `Kafka v0.8.0`

Logging separate issues for the above failed cases.

- The other mentioned apps in the PR, deploy on a k8s 1.16 cluster. 
-  Tested on rancher version : `dramich/rancher:dev`

**Comment by sowmyav27** [comment]
**Reproduced the issue on `v2.3.1` rancher:**

**Steps:**

1. Go to Global - Catalogs, add a catalog from `https://github.com/gitlawr/helm-charts.git`, `branch: k8s-1.16`
2. Install mysql app with persistence.enabled=false answers to the Default project. Wait till the app is installed and active.
3. Delete the app.
4. The workload does get deleted


**Verified the issue on `2.3-head` - latest**
**Steps:**
1. Go to Global - Catalogs, add a catalog from `https://github.com/gitlawr/helm-charts.git`, `branch: k8s-1.16`
2. Install mysql app with persistence.enabled=false answers to the Default project. Wait till the app is installed and active.
3. Delete the app.
4. The workload also is deleted.

Logs show no warnings:
```
[main] 2019/10/17 20:29:12 Starting Tiller v2.14+unreleased (tls=false)
[main] 2019/10/17 20:29:12 GRPC listening on :44878
[main] 2019/10/17 20:29:12 Probes listening on :58891
[main] 2019/10/17 20:29:12 Storage driver is ConfigMap
[main] 2019/10/17 20:29:12 Max history per release is 10
[storage] 2019/10/17 20:29:13 getting release history for "mysql"
[tiller] 2019/10/17 20:29:13 uninstall: Release not loaded: mysql
[main] 2019/10/17 20:29:20 Starting Tiller v2.14+unreleased (tls=false)
[main] 2019/10/17 20:29:20 GRPC listening on :53845
[main] 2019/10/17 20:29:20 Probes listening on :37358
[main] 2019/10/17 20:29:20 Storage driver is ConfigMap
[main] 2019/10/17 20:29:20 Max history per release is 10
[storage] 2019/10/17 20:29:21 getting release history for "mysql"
[tiller] 2019/10/17 20:29:21 uninstall: Deleting mysql
[tiller] 2019/10/17 20:29:21 executing 1 pre-delete hooks for mysql
[tiller] 2019/10/17 20:29:21 hooks complete for pre-delete mysql
[storage] 2019/10/17 20:29:21 updating release "mysql.v1"
[kube] 2019/10/17 20:29:22 Starting delete for "mysql" Service
[kube] 2019/10/17 20:29:22 Starting delete for "mysql" Deployment
[kube] 2019/10/17 20:29:22 Starting delete for "mysql-test" ConfigMap
[kube] 2019/10/17 20:29:23 Starting delete for "mysql" Secret
[tiller] 2019/10/17 20:29:23 executing 1 post-delete hooks for mysql
[tiller] 2019/10/17 20:29:23 hooks complete for post-delete mysql
[tiller] 2019/10/17 20:29:23 purge requested for mysql
[storage] 2019/10/17 20:29:23 deleting release "mysql.v1"
2019/10/17 20:29:23 [WARNING] release "mysql" deleted
```
